src/stdlib/File.ts

```diff
@@ -0,0 +1,87 @@
+import { Callable, ValueKind, BrsString, BrsBoolean } from "../brsTypes";
+import { Interpreter } from "../interpreter";
+import { Volume } from "memfs/lib/volume";
+
+/* Returns the brs-path uri scheme or an empty string if not found */
+export function parseVolumeName(path: string): string {
+    const volumeRegex = new RegExp("^([a-z]+):");
+    const volumePrefix = path.match(volumeRegex);
+    if (volumePrefix !== null) {
+        return volumePrefix[1];
+    } else {
+        return "";
+    }
+}
+
+/*
+ * Returns a memfs volume based on the brs path uri.  For example, passing in
+ * "tmp:///test.txt" will return the memfs temporary volume on the interpreter. 
+ * 
+ * Returns invalid in no appopriate volume is found for the path
+ */
+export function getVolumeByPath(interpreter: Interpreter, path: string): Volume | null {
+    const volumePrefix = parseVolumeName(path);
```

Comment:
Consider using `node`'s built-in `URL` class instead of writing this parser yourself - it'll save us some trouble and we won't have to test the implementation :smile: Something like this would probably work:

```
try {
    let url = new URL(path);
    if (url.scheme === "tmp:") return interpreter.temporaryVolume;
catch (err) {
    return null;
}
```

Reply:
Thanks. Switched to this, which also handles unknown volumes:

```
    try {
        const protocol = new URL(path).protocol;
        if (protocol === "tmp:") return interpreter.temporaryVolume;
    } catch (err) {
        return null;
    }
    return null;
```